### DISCUSSION TITLE
Why user channel and role is not assign at the time of user signup?

### DISCUSSION THREAD

**Discussion by komalm** [Sunbird-Lern/Community#76] [Issues]
Hello @reshmi-nair, @rhwarrier  

Please assist us with the following problem.

**Site Environments Details** - SunbirdEd portal of version 4.4.1

I've used the signup process to create a new user. Additionally, a newly created user must be assigned to a new channel at the time of signup (not default channel). Now, when I looked at the user information, I discovered that the channel data was set to the default value of one (rootorg)

`API - {{host}}/learner/user/v2/signup`

 ```
 "request": {
        "firstName": "test17",
        "password": "Password@123",
        "dob": "1993",
        "email": "test17@gmail.com",
        "emailVerified": true,
        "reqData": "{\"iv\":\sdfsdf"\",\"encryptedData\":\"sdsdsdfs\"}",
        "channel": "new-channel",
        "roles": [
            "CONTENT_CREATOR"
        ]
    }
```

Note: There was not a single error when creating the above user or using the above API.

**Result**

The user gives the default channel and roles an empty array after creating a user with the above API.

CC @renuka-S, @vinod303, @Krishnaj20 

`API - {{host}}/learner/user/v2/signup`

![NewUserPayload png (821×300)](https://user-images.githubusercontent.com/3881053/218733039-44ab6a72-c6ee-45ef-96e9-35a493503026.png)

**After Login User Detail API** - Assign default channel/root channel and Roles are empty.

![newUser1 png (1042×394) png (977×377)](https://user-images.githubusercontent.com/3881053/218733905-172b5f31-c5a3-454f-b11b-71e737b619fc.png)

![newUser2](https://user-images.githubusercontent.com/3881053/218733201-7209f2c3-51cd-4257-a8b6-3fddb56e0e44.png)

**Comment by reshmi-nair** [comment]
@komalm Pls use /v1/ssouser/create API if you want to provide a specific channel while creating user. Signup API will always create user under default organisation.

**Comment by AmiableAnil** [comment]
Closing this thread
